pkg/agent/agentclient/client.go

```diff
@@ -148,6 +158,24 @@ func (a *AgentClient) pipe(conn net.Conn, connID int64) {
 				glog.Warningf("stream send error: %v", err)
 			}
 		}
+	}
+}
+
+func (a *AgentClient) proxyToRemote(conn net.Conn, dataCh <-chan []byte, cleanup func()) {
+	defer cleanup()
 
+	for d := range dataCh {
+		pos := 0
+		for {
+			n, err := conn.Write(d[pos:])
+			if n > 0 {
```

Comment:
I agree that n should be zero if error. That having been said if we end up with a positive n and an error, I want us to deal with the error. Can we please move that check first?

Reply:
if 0 < n < len(d), it is likely to be some transient error, and we can update buffer and try again, if something really goes wrong, we deal with it there. but if n == len(d). we should simply break. Yep. I'll update that.